nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/converter/ProtobufDataConverter.java

```diff
@@ -368,4 +410,30 @@ private RecordSchema generateRecordSchema(String typeName) {
     private String getQualifiedTypeName(String typeName) {
         return typeName.substring(typeName.lastIndexOf('/') + 1);
     }
+
+    private <T> List<T> processRepeatedValues(CodedInputStream input, ValueReader<CodedInputStream, T> valueReader) {
+        List<T> result = new ArrayList<>();
+        try {
+            while (input.getBytesUntilLimit() > 0) {
+                result.add(valueReader.apply(input));
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException("Unable to parse repeated field");
+        }
+        return result;
+    }
+
+    private boolean isLengthDelimitedType(ProtoField protoField) {
+        final ProtoType protoType = protoField.getProtoType();
+        if (schema.getType(protoType) instanceof MessageType) {
+            return true;
+        }
+
+        if (protoType.isScalar()) {
+            FieldType fieldType = FieldType.findValue(protoType.getSimpleName());
+            return fieldType.equals(STRING) || fieldType.equals(BYTES);
+        } else {
+            return false;
+        }
```

Comment:
Recommend refactoring this method slightly to have a single `return` statement. The return value could be set in the appropriate conditional, making it `final` in the initial declaration.

Reply:
I've refactored the method, let me know what do you think about it.